mat/triangular.go

```diff
@@ -452,15 +452,49 @@ func (t *TriDense) MulTri(a, b Triangular) {
 		defer restore()
 	}
 
-	// TODO(btracey): Improve the set of fast-paths.
+	var aDiag, bDiag bool
```

Comment:
I don't think we need to check the bandwidths. If it says it's diagonal, we should feel free to assume it's diagonal.